### PR TITLE
Implement a user profile in the desktop client

### PR DESCRIPTION
**Pull request by Vladyslav-Kuksiuk**
This PR implements a user profile info tab in the desktop client and modal dialogs.

Here are some usage scenarios:
- Chat deletion and creation via profile:
chat deletion and creation_Trim
- Chat deletion via the 'More' button:
ChatSPN 2023-06-22 13-44-13_Trim
- Log out:
ChatSPN 2023-06-22 13-47-58_Trim

### CODE REVIEW

**Review by Artem-Semenov-dev — COMMENTED** [review]
LGTM.

**Review by armiol — CHANGES_REQUESTED** [review]
General UI idea is to have less pop-ups, so that users would not have any blocking windows unless something important happens (payment, deletion, etc.).

Please have a look at other chats, and analyse how they implement similar functionality.

I am not reviewing the code changes now, because the approach in general should be different.

File: client/src/main/kotlin/io/spine/examples/chatspn/desktop/ChatsPage.kt [hunk 1]
```diff
@@ -475,17 +506,258 @@ private fun ChatNotChosenBox() {
...
+                .fillMaxWidth()
+                .padding(12.dp, 8.dp),
+            verticalArrangement = Arrangement.Center
+        ) {
+            Row(verticalAlignment = Alignment.CenterVertically) {
+                Icon(
+                    Icons.Default.Email,
+                    "Email",
+                    Modifier.size(20.dp),
+                    tint = MaterialTheme.colorScheme.primary
```
Comment (Artem-Semenov-dev) [inline client/src/main/kotlin/io/spine/examples/chatspn/desktop/ChatsPage.kt:632]: Please choose one style: write the parameters' name or not.

File: client/src/main/kotlin/io/spine/examples/chatspn/desktop/ChatsPage.kt [hunk 2]
```diff
@@ -475,17 +506,258 @@ private fun ChatNotChosenBox() {
...
+ * Represents a button on an info tab.
+ */
+@Composable
+private fun InfoTabButton(
+    text: String = "",
+    icon: ImageVector? = null,
+    contentColor: Color = MaterialTheme.colorScheme.primary,
+    onClick: () -> Unit
+) {
+    Box(Modifier.clip(RoundedCornerShape(8.dp))) {
```
Comment (Artem-Semenov-dev) [inline client/src/main/kotlin/io/spine/examples/chatspn/desktop/ChatsPage.kt:661]: I think we can extract this to `MaterialTheme` for more readability.

File: client/src/main/kotlin/io/spine/examples/chatspn/desktop/ChatsPage.kt [hunk 3]
```diff
@@ -495,17 +767,183 @@ private fun ChatTopbar(model: ChatsPageModel) {
...
-            verticalAlignment = Alignment.CenterVertically
+        content()
+    }
+}
+
+/**
+ * Represents the 'More' button for the chat.
+ */
+@Composable
+private fun ChatMoreButton(model: ChatsPageModel, chat: ChatData) {
```
Comment (Artem-Semenov-dev) [inline client/src/main/kotlin/io/spine/examples/chatspn/desktop/ChatsPage.kt:778]: Do we need a `Chat` prefix here? Consider renaming.

File: client/src/main/kotlin/io/spine/examples/chatspn/desktop/ChatsPage.kt [hunk 4]
```diff
@@ -495,17 +767,183 @@ private fun ChatTopbar(model: ChatsPageModel) {
...
+                    }
+                    isVisible.value = false
+                }
+            }
+        }
+    }
+}
+
+/**
+ * Represents the text button.
```
Comment (Artem-Semenov-dev) [inline client/src/main/kotlin/io/spine/examples/chatspn/desktop/ChatsPage.kt:915]: I think this deserves better documentation.

**Review by Artem-Semenov-dev — COMMENTED** [review]
@Vladyslav-Kuksiuk LGTM with a few comments.

File: client/src/main/kotlin/io/spine/examples/chatspn/desktop/ChatsPage.kt [hunk 5]
```diff
@@ -475,17 +506,258 @@ private fun ChatNotChosenBox() {
 }
 
 /**
- * Represents the topbar of chat content.
+ * Represents the info tab with the user profile.
+ */
+@Composable
+private fun ProfileTab(model: ChatsPageModel) {
```
Comment (armiol) [inline client/src/main/kotlin/io/spine/examples/chatspn/desktop/ChatsPage.kt:512]: I think, profiles should belong to their own Kotlin file. 

It's weird that they are a part of chats, even though at the moment these features are somewhat related in navigation.

**Review by armiol — CHANGES_REQUESTED** [review]
LGTM in general. But please see my comment. 

Also, let's not use any icons from existing commercial applications. We should be very careful about licensing. Right now some of the icons look very telegram-y.

File: client/src/main/kotlin/io/spine/examples/chatspn/desktop/navigation/NavigationModel.kt [hunk 6]
```diff
@@ -0,0 +1,301 @@
...
+ * Pages in the application.
+ */
+public enum class Page {
+    CHAT, LOGIN, REGISTRATION, PROFILE
+}
+
+/**
+ * Data for the chat preview.
+ */
+public data class ChatData(
```
Comment (armiol) [inline client/src/main/kotlin/io/spine/examples/chatspn/desktop/navigation/NavigationModel.kt:212]: With the documented purpose in mind, this name is too generic, especially for `public` data type.

File: client/src/main/kotlin/io/spine/examples/chatspn/desktop/profile/ChatDeletionDialog.kt [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import io.spine.examples.chatspn.chat.Chat
+import io.spine.examples.chatspn.desktop.component.ModalWindow
+import io.spine.examples.chatspn.desktop.component.TextButton
+import io.spine.examples.chatspn.desktop.navigation.ChatData
+import kotlinx.coroutines.launch
+
+/**
+ * Representation of the chat deletion confirmation modal.
```
Comment (armiol) [inline client/src/main/kotlin/io/spine/examples/chatspn/desktop/profile/ChatDeletionDialog.kt:56]: "A modal dialog allowing to delete some chat."

File: client/src/main/kotlin/io/spine/examples/chatspn/desktop/profile/LogoutDialog.kt [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.dp
+import io.spine.examples.chatspn.desktop.component.ModalWindow
+import io.spine.examples.chatspn.desktop.component.TextButton
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+/**
+ * Returns a representation of the logout confirmation modal.
```
Comment (armiol) [inline client/src/main/kotlin/io/spine/examples/chatspn/desktop/profile/LogoutDialog.kt:52]: Can we please not use "represent" and its derivatives this often?

This word has many meanings, and is way too formal. Also, [this](https://constitutioncenter.org/blog/no-taxation-without-representation#:~:text=The%20Stamp%20Act%20Congress%20met,lacked%20representation%20in%20British%20Parliament.).

@Artem-Semenov-dev FYI.

Comment (armiol) [inline client/src/main/kotlin/io/spine/examples/chatspn/desktop/profile/LogoutDialog.kt:52]: "A modal dialog allowing the current user to log out."

**Review by armiol — CHANGES_REQUESTED** [review]
@Vladyslav-Kuksiuk mostly LGTM. Please see my comments.

File: client/src/main/kotlin/io/spine/examples/chatspn/desktop/navigation/NavigationModel.kt (hunk 6 above)
Comment (armiol) [reply client/src/main/kotlin/io/spine/examples/chatspn/desktop/navigation/NavigationModel.kt:218]: This is still too broad. Please rename more precisely.

And its documentation was fine.

**Review by armiol — CHANGES_REQUESTED** [review]

File: client/src/main/kotlin/io/spine/examples/chatspn/desktop/Client.kt [hunk 9]
```diff
@@ -385,6 +438,44 @@ public class DesktopClient(
         return subscription
     }
 
+    /**
+     * Observes the outcome of the command.
+     *
+     * When a success or failure event is emitted, subscriptions will be canceled.
```
Comment (armiol) [inline client/src/main/kotlin/io/spine/examples/chatspn/desktop/Client.kt:444]: "are cancelled"

Please DO have a read on English conditionals.

File: client/src/main/kotlin/io/spine/examples/chatspn/desktop/navigation/NavigationModel.kt [hunk 10]
```diff
@@ -0,0 +1,259 @@
...
+                        (chatCard.memberList[0].id.equals(user) ||
+                                chatCard.memberList[1].id.equals(user))
+            }
+        return chat
+    }
+
+    /**
+     * Opens a page with a user profile.
+     *
+     * @param userId id of the user to open profile
```
Comment (armiol) [inline client/src/main/kotlin/io/spine/examples/chatspn/desktop/navigation/NavigationModel.kt:168]: `ID`

Please do a case-sensitive search for "id" in the documentation.

File: client/src/main/kotlin/io/spine/examples/chatspn/desktop/navigation/NavigationModel.kt [hunk 11]
```diff
@@ -0,0 +1,259 @@
...
+    public fun openUserProfile(userId: UserId) {
+        val user = client.findUser(userId)
+        profilePageState.userProfile.value = user ?: UserProfile.getDefaultInstance()
+        profilePageState.chatState.value =
+            if (null == user) null else findPersonalChat(user.id)
+        currentPage.value = Page.PROFILE
+    }
+
+    /**
+     * Opens a page with chat info. If the chat is personal, a user profile will be opened.
```
Comment (armiol) [inline client/src/main/kotlin/io/spine/examples/chatspn/desktop/navigation/NavigationModel.kt:179]: The first paragraph should always be a single sentence.

File: client/src/main/kotlin/io/spine/examples/chatspn/desktop/navigation/NavigationModel.kt [hunk 12]
```diff
@@ -0,0 +1,259 @@
...
+/**
+ * List of `ChatData`.
+ */
+public typealias ChatList = List<ChatCard>
+
+/**
+ * Finds chat in the list by ID.
+ *
+ * @param id ID of the chat to find
+ * @return found chat or `null` if chat is not found
```
Comment (armiol) [inline client/src/main/kotlin/io/spine/examples/chatspn/desktop/navigation/NavigationModel.kt:238]: "chat with the given ID, or `null` if ..."

**Review by armiol — APPROVED** [review]
@Vladyslav-Kuksiuk LGTM with some comments to address.

**Pull request merged**